QWeb/internal/screenshot.py

```diff
@@ -147,6 +147,7 @@ def _draw_contours(diff: ndarray, ref_image_c: ndarray) -> ndarray:
     index = 1 if cv2.__version__.startswith("3") else 0  # type: ignore
 
     for c in contours[index]:
+        assert isinstance(c, ndarray)
```

Comment:
Is  `assert` intended for debugging or runtime checks?

If this is intended for active runtime check and someone builds QWeb from sources with `-O` flag then this line will be disabled and can lead to unexpected situation.

https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement

Reply:
If this is for runtime check then the line should be changed to something like (mimicked from the py doc)
`if not isinstance(c, ndarray): raise AssertationError`

Reply:
ah, yes. I will fix this with a new commit. I will actually then add that notification functionality in this pr too (just to save time and test rounds), so maybe I'll mark this as a draft for a while.

Reply:
Ok, I don't seem to be able to put this back as draft. In any case this was there just to make typing happy, forgot to make the final change there. There was a bit of a change of types in latest cv2 version. We also still supported cv2 version 3 (very old) which had also differences. I now simplified this to make it match the signature/types of minimum cv2 expected version we require in deps:

`contours, _ = cv2.findContours(thresh.copy(), cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)`

Reply:
fixed with a new commit

Reply:
doh, fixed all the reasons for needing that assert for testing in the first place, but forgot to touch the assert itself